### PR TITLE
Remove mentions of `controller..can` choice syntax in 2.8.0 docs

### PR DESCRIPTION
**Pull request by akrmn**
Part of https://github.com/digital-asset/daml/issues/17361

Daml PRs: https://github.com/digital-asset/daml/pull/17362, https://github.com/digital-asset/daml/pull/17390
DA-GHC PR: https://github.com/digital-asset/ghc/pull/171
docs.daml.com PR: https://github.com/digital-asset/docs.daml.com/pull/453

## TODO

- [x] for this to work, the `daml` version in `docs/2.8.0/versions.json` must be updated to the snapshot with the changes from https://github.com/digital-asset/daml/pull/17390
    https://github.com/digital-asset/docs.daml.com/blob/d3fb57ea66aee8d3f4b57942e17ed4535f851df9/docs/2.8.0/versions.json#L2

### CODE REVIEW

File: docs/2.8.0/docs/daml/reference/choices.rst
Comment (akrmn) [inline docs/2.8.0/docs/daml/reference/choices.rst:17]: these snippets will likely fail until the daml PR is merged into `main`

File: docs/2.6.5/docs/daml/reference/choices.rst
Comment (akrmn) [inline docs/2.6.5/docs/daml/reference/choices.rst:59]: I don't think this has ever been true, at least not since I joined

Comment (garyverhaegen-da) [reply docs/2.6.5/docs/daml/reference/choices.rst:59]: AFAIK It's true within a file/module, but not across a project.

Comment (akrmn) [reply docs/2.6.5/docs/daml/reference/choices.rst:59]: thanks, I've updated to reflect this

File: docs/2.8.0/docs/daml/reference/choices.rst
Comment (carrielaben-da) [inline docs/2.8.0/docs/daml/reference/choices.rst:40]:
```suggestion
  You **must** make sure that the controller parties are observers (or signatories) of the contract, otherwise they cannot see the contract (and therefore cannot exercise the choice).
```

File: docs/2.8.0/docs/daml/reference/choices.rst
Comment (carrielaben-da) [inline docs/2.8.0/docs/daml/reference/choices.rst:47]:
```suggestion
*Choice observers* can be attached to a choice using the ``observer`` keyword. The choice observers are a list of parties who are not stakeholders but who see all the consequences of the action.
```

**Review by carrielaben-da — APPROVED** [review]
LGTM

**Comment by akrmn** [comment]
IIUC, this will only build successfully once https://github.com/digital-asset/daml/pull/17362 (https://github.com/digital-asset/daml/commit/1bee52e40c889d4cbf4cc95b664d0b354cb6c82c) is released and we update `2.8.0/versions.json` to match:

https://github.com/digital-asset/docs.daml.com/blob/52d0334f4ada31588d17bb443ba49982436e11a8/docs/2.8.0/versions.json#L2

on the flipside, that means we don't really need to keep the old snippets around, so I'll be removing them next week

**Comment by akrmn** [comment]
Updated `2.8.0/versions.json:daml` to https://github.com/digital-asset/daml/releases/tag/v2.8.0-snapshot.20230908.12101.0.v1bee52e4

File: docs/2.8.0/versions.json
Comment (akrmn) [inline docs/2.8.0/versions.json:2]: I messed up, https://github.com/digital-asset/daml/pull/17390 isn't part of today's release. do I have to wait for the next release or is it okay to point to (the current commit of) `main`?

Comment (garyverhaegen-da) [inline docs/2.8.0/versions.json:2]:
```suggestion
  "daml": "2.8.0-snapshot.20230912.12110.0.v788fccd9",
```
These must match folders on Artifactory, so using git commits won't work here.

**Review by carrielaben-da — APPROVED** [review]

File: docs/2.8.0/versions.json
Comment (moisesackerman-da) [reply docs/2.8.0/versions.json]: ah, I didn't see this last week and thought I had to wait until a snapshot _release_, now I see that the snapshots on Artifactory are enough. Oh well, lesson learned for the next time :)

**Pull request merged**
